src/lock/script.rs

```diff
@@ -10,13 +11,17 @@ struct ExternalData<'a> {
     name: &'a str,
     dir: &'a str,
     files: Vec<&'a str>,
+    hooks: &'a BTreeMap<String, String>,
 }
 
 impl LockedConfig {
     /// Generate the script.
     pub fn script(&self, ctx: &Context) -> Result<String> {
         // Compile the templates
         let mut engine = upon::Engine::new();
+        engine.add_filter("contains", |map: BTreeMap<String, _>, key: String| {
```

Comment:
What do you think of this idea to make the template cleaner.

- The following makes use of the fact that `None` and `Value::None` are rendered as an empty string
- It also takes the map by reference to avoid cloning
- It instead uses two filters, one to lookup the value and one to append a newline, this allows us to avoid an `if` statement in the template.

```rust
        engine.add_filter(
            "get",
            |map: &BTreeMap<String, upon::Value>, key: &str| -> Option<upon::Value> {
                map.get(key).cloned()
            },
        );

        engine.add_filter("nl", |mut v: upon::Value| -> upon::Value {
            if let upon::Value::String(s) = &mut v {
                if !s.ends_with('\n') {
                    s.push('\n');
                }
            }
            v
        });
```

And then use the following in the templates

```
{{ hooks | get: "pre" | nl }} ... {{ hooks | get: "pre" | nl }}
```

Reply:
Makes sense. Thanks for the review. I committed the changes as you suggested.